File: omenergy/src/main/kotlin/io/omenergy/omenergy/service/TrainingDescriptionService.kt
package io.omenergy.omenergy.service

import io.omenergy.omenergy.dto.TrainingDescriptionDto
import io.omenergy.omenergy.dto.toEntity
import io.omenergy.omenergy.entity.TrainingDescriptionData
import io.omenergy.omenergy.repository.TrainingDescriptionDataRepository
import org.springframework.stereotype.Service

@Service
class TrainingDescriptionService(val trainingDescriptionDataRepository: TrainingDescriptionDataRepository) {

    fun defaultData(): List<TrainingDescriptionData> {
        return listOf(
            TrainingDescriptionData(
                header = "Cardio Training . Fake data",
                description = "A elementum ligula lacus ac quam ultrices a scelerisque praesent vel suspendisse scelerisque a aenean hac montes.",
                image = "images/icons/heart-blue.png"
            ),
            TrainingDescriptionData(
                header = "Yoga Pilates . Fake data",
                description = "A elementum ligula lacus ac quam ultrices a scelerisque praesent vel suspendisse scelerisque a aenean hac montes.",
                image = "images/icons/guru-blue.png"
            ),
            TrainingDescriptionData(
                header = "Power Training . Fake data",
                description = "A elementum ligula lacus ac quam ultrices a scelerisque praesent vel suspendisse scelerisque a aenean hac montes.",
                image = "images/icons/weight-blue.png"
            )
        )
    }

    fun getData() : List<TrainingDescriptionDto> = trainingDescriptionDataRepository.findAll().map { mapToDto(it) }

    fun getDataByHeader(header: String) : List<TrainingDescriptionDto> = trainingDescriptionDataRepository.findByHeader(header).map { mapToDto(it) }


    fun resetToDefault() : Boolean {
        return  trainingDescriptionDataRepository.save(defaultData())
    }

    fun save(trainingDescriptionData: TrainingDescriptionDto) : Boolean {
      return  trainingDescriptionDataRepository.save(trainingDescriptionData.toEntity())
    }

    fun update(trainingDescriptionData: TrainingDescriptionDto) : Boolean {

        return  trainingDescriptionDataRepository.update(trainingDescriptionData.toEntity())
    }

    fun mapToDto(data: TrainingDescriptionData) = TrainingDescriptionDto(
        id = data.id.toString(),
        header = data.header,
        description = data.description,
        image = data.image?: ""
    )

    fun getDataById(id: String) = mapToDto(trainingDescriptionDataRepository.findOne(id))

}

File: taffeite-underlying-service/services/TrainingDescriptionService.go
package services

import "taffeite.com/taffeite-underlying-service/dto"

func GetTrainingDescriptions() []dto.TrainingDescriptionDto {

	return []dto.TrainingDescriptionDto{
		{"1", "Убрать лишний вес",
			"A elementum ligula lacus ac quam ultrices a scelerisque praesent vel suspendisse scelerisque a aenean hac montes.",
			"assets/images/icons/guru-blue.png"},
		{"2", "Раскрыть женскую сексуальность",
			"A elementum ligula lacus ac quam ultrices a scelerisque praesent vel suspendisse scelerisque a aenean hac montes.",
			"assets/images/icons/guru-blue.png"},

		{"3", "Научиться танцевать",
			"A elementum ligula lacus ac quam ultrices a scelerisque praesent vel suspendisse scelerisque a aenean hac montes.",
			"assets/images/icons/guru-blue.png"},
		{"4", "Эффективные тренировки без спортзала",
			"A elementum ligula lacus ac quam ultrices a scelerisque praesent vel suspendisse scelerisque a aenean hac montes.",
			"assets/images/icons/guru-blue.png"},
		{"5", "Победить депрессию и апатию",
			"A elementum ligula lacus ac quam ultrices a scelerisque praesent vel suspendisse scelerisque a aenean hac montes.",
			"assets/images/icons/guru-blue.png"},

		{"6", "Стать богатым, спокойным и знаменитым",
			"A elementum ligula lacus ac quam ultrices a scelerisque praesent vel suspendisse scelerisque a aenean hac montes.",
			"assets/images/icons/guru-blue.png"},
	}
}
